File: web/lib/features/superadmin/domain/platform_config.dart
/// Réglages globaux de la plateforme : abonnement (prix, essai) et email
/// SMTP dynamique (Paramètres → Email, super-admin) — voir
/// `core/notifications.py::send_email` côté backend.
class PlatformConfig {
  final int abonnementMontantHtg;
  // Prix du PROCHAIN renouvellement, distinct du prix courant — null si
  // aucun changement de prix n'est annoncé (le renouvellement se fait alors
  // au même prix que abonnementMontantHtg).
  final int? abonnementRenouvellementHtg;
  final int essaiJours;
  final String? smtpHost;
  final int smtpPort;
  final String? smtpUser;
  // Jamais le mot de passe réel — seulement s'il est défini ou non.
  final bool smtpPasswordDefini;
  final String? smtpFromEmail;

  const PlatformConfig({
    required this.abonnementMontantHtg,
    required this.abonnementRenouvellementHtg,
    required this.essaiJours,
    required this.smtpHost,
    required this.smtpPort,
    required this.smtpUser,
    required this.smtpPasswordDefini,
    required this.smtpFromEmail,
  });

  factory PlatformConfig.fromJson(Map<String, dynamic> json) => PlatformConfig(
        abonnementMontantHtg: json['abonnement_montant_htg'] as int,
        abonnementRenouvellementHtg: json['abonnement_renouvellement_htg'] as int?,
        essaiJours: json['essai_jours'] as int,
        smtpHost: json['smtp_host'] as String?,
        smtpPort: json['smtp_port'] as int,
        smtpUser: json['smtp_user'] as String?,
        smtpPasswordDefini: json['smtp_password_defini'] as bool,
        smtpFromEmail: json['smtp_from_email'] as String?,
      );
}

File: backend/app/models/platform_config.py
import uuid

from sqlmodel import Field, SQLModel


class PlatformConfig(SQLModel, table=True):
    """Réglages globaux de la plateforme — une seule ligne (singleton),
    lue/écrite par le super admin. Extensible à d'autres réglages plus tard."""

    __tablename__ = "platform_config"

    id: str = Field(default_factory=lambda: str(uuid.uuid4()), primary_key=True)
    abonnement_montant_htg: int = Field(default=100)
    # Prix appliqué au PROCHAIN renouvellement — distinct de
    # abonnement_montant_htg pour annoncer un changement de prix à l'avance
    # (visible sur la page Abonnement du client, à côté de la date de
    # renouvellement) avant qu'il ne soit effectivement facturé. None =
    # aucun changement annoncé, le renouvellement se fait au même prix que
    # abonnement_montant_htg (voir abonnement.py::_montant_renouvellement).
    abonnement_renouvellement_htg: int | None = Field(default=None)
    # Durée de la période d'essai gratuit (PRD §8.5), en jours à partir de
    # `Entreprise.date_creation` — voir `_abonnement_actif` (transactions.py).
    essai_jours: int = Field(default=14)

    # Configuration email SMTP dynamique — éditable par le super-admin
    # (Paramètres → Email), lue par core/notifications.py::send_email() au
    # lieu des valeurs statiques .env (settings.SMTP_*, gardées en repli si
    # ces champs sont vides). Même principe que PlatformConfig.smtp_* de
    # pos_api : source de vérité en base, pas dans la config de déploiement.
    smtp_host: str | None = Field(default=None)
    smtp_port: int = Field(default=587)
    smtp_user: str | None = Field(default=None)
    smtp_password: str | None = Field(default=None)
    smtp_from_email: str | None = Field(default=None)
